### PR TITLE
fixing data validation for Combobox control and all SelectingItemsControl-s 

### LINKED ISSUE #5652: ComboBox validation not working for SelectedItem
**Describe the bug**

Data validation errors are not displayed when binding SelectedItem of ComboBox.

A common usage case for ComboBox is to bind the SelectedItem and Items properties to the view model.

**To Reproduce**

Bind a view model property to SelectedItem and make the property setter throw and exception.
The UI does not display the error.

**Expected behavior**

A data validation error for SelectedItem should be displayed in the UI.

**Screenshots**

**Desktop (please complete the following information):**

- OS: Windows
- Version 0.10.0

**Additional context**

Related to #3905, #4054 and #5586

### PR DESCRIPTION
**Pull request by aljosas**
## What does the pull request do?
Adds data validation support to the SelectedItemsControl.SelectedItemProperty.

## What is the current behavior?
Data validation errors for the SelectedItem property do not update the UI.

## What is the updated/expected behavior with this PR?
Data validation errors for the SelectedItem property will update the UI.

## How was the solution implemented (if it's not obvious)?
Used DataValidation from Binding.

## Checklist

- [X ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Fixed issues

Fixes #5652

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5653) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5653) before we can accept your contribution.<br/><br/>:x: aljosas [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5653)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=5653) it.</sub>

**Comment by maxkatz6** [comment]
Hm, it seems we support data validation for ComboBox in the default/fluent templates, but errors are not updated there on value changed. Before this PR.
Thanks.

Note, that our list box template doesn't contain DataValidationError component to display errors. I also doubt if it should contain, it's not clear how it should look like. Although with this PR users can implement this template if they want to.

File: tests/Avalonia.Controls.UnitTests/CarouselTests.cs [hunk 1]
```diff
@@ -16,12 +20,7 @@ public void First_Item_Should_Be_Selected_By_Default()
         {
             var target = new Carousel
             {
-                Template = new FuncControlTemplate<Carousel>(CreateTemplate),
-                Items = new[]
-                {
-                    "Foo",
-                    "Bar"
-                }
+                Template = new FuncControlTemplate<Carousel>(CreateTemplate), Items = new[] { "Foo", "Bar" }
```
Comment (maxkatz6) [inline tests/Avalonia.Controls.UnitTests/CarouselTests.cs:23]: Can you revert all of these formatting changes combining everything into one line? Here and below. There is also one place in SelectingItemsControl.

File: tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs [hunk 2]
```diff
@@ -173,5 +176,67 @@ public void Detaching_Closed_ComboBox_Keeps_Current_Focus()
...
+        
+        private TextBox GetTextBox(ComboBox control)
+        {
+            return control.GetTemplateChildren()
+                // .OfType<ButtonSpinner>()
+                // .Select(b => b.Content)
+                .OfType<TextBox>()
+                .First();
+        }
+        // private IControlTemplate CreateTemplate()
```
Comment (maxkatz6) [inline tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs:225]: Is it WIP or can you remove this commented code?

File: tests/Avalonia.Controls.UnitTests/ListBoxTests.cs [hunk 3]
```diff
@@ -559,5 +559,12 @@ public Item(string value)
 
             public string Value { get; }
         }
+
+
+        [Fact]
+        public void SelectedItem_Validation()
```
Comment (maxkatz6) [inline tests/Avalonia.Controls.UnitTests/ListBoxTests.cs:565]: Unused test?

**Review by maxkatz6 — CHANGES_REQUESTED** [review]

File: tests/Avalonia.Controls.UnitTests/ListBoxTests.cs (hunk 3 above)
Comment (aljosas) [reply tests/Avalonia.Controls.UnitTests/ListBoxTests.cs:565]: I didn't see that there was empty test, added code in fix PR

File: tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs (hunk 2 above)
Comment (aljosas) [reply tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs:225]: I will remove the code

File: tests/Avalonia.Controls.UnitTests/CarouselTests.cs (hunk 1 above)
Comment (aljosas) [reply tests/Avalonia.Controls.UnitTests/CarouselTests.cs:23]: I will fix it.

**Comment by maxkatz6** [comment]
@aljosas can you check failing test?

**Comment by aljosas** [comment]
> @aljosas can you check failing test?

Hi, I fixed failing tests

**Review by maxkatz6 — APPROVED** [review]

**Comment by maxkatz6** [comment]
@aljosas sorry for delay, thanks!

**Comment by maxkatz6** [comment]
@aljosas did you closed PR by mistake?

**Comment by aljosas** [comment]
> @aljosas did you closed PR by mistake?

Yes, I checked and found out my kids were playing with my phone :).

**Pull request merged**